metaflow/plugins/kfp/tests/run_integration_tests.py

```diff
@@ -51,10 +58,53 @@ def obtain_flow_file_paths(flow_dir_path: str) -> List[str]:
         and not file_name.startswith(".")
         and not "raise_error_flow" in file_name
         and not "accelerator_flow" in file_name
+        and not "s3_sensor_flow" in file_name
+        and not "upload_to_s3_flow" in file_name
     ]
     return file_paths
 
 
+def test_s3_sensor_flow(pytestconfig) -> None:
+    # ensure the s3_sensor waits for some time before the key exists
+    random_string = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(15))
```

Comment:
you can create a UID to ensure it is unique

Reply:
Cool, I'll switch over to using the `import uuid` library.